### PR TITLE
Test on Node 18

### PR DESCRIPTION
**Pull request by rekmarks**
Starts testing the module template on Node 18 in CI, and updates dependencies for Node 18 compatibility.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -37,7 +37,7 @@
     "eslint-config-prettier": "^8.1.0",
     "eslint-plugin-import": "^2.22.1",
     "eslint-plugin-jest": "^24.3.4",
-    "eslint-plugin-jsdoc": "^36.1.0",
+    "eslint-plugin-jsdoc": "^39.2.9",
```
Comment (rekmarks) [inline package.json:40]: This is the minimum version of the package that's compatible with Node 18. They make no note of this change in their changelog.

File: package.json [hunk 2]
```diff
@@ -29,23 +29,23 @@
...
+    "@types/jest": "^28.1.6",
     "@types/node": "^17.0.23",
     "@typescript-eslint/eslint-plugin": "^4.21.0",
     "@typescript-eslint/parser": "^4.21.0",
     "eslint": "^7.23.0",
     "eslint-config-prettier": "^8.1.0",
     "eslint-plugin-import": "^2.22.1",
     "eslint-plugin-jest": "^24.3.4",
-    "eslint-plugin-jsdoc": "^36.1.0",
+    "eslint-plugin-jsdoc": "^39.2.9",
```
Comment (Gudahtt) [inline package.json:40]: This is outside the range requested by our ESLint config: https://github.com/MetaMask/eslint-config/blob/main/packages/base/package.json#L41

Ideally we'd update our config first

File: package.json [hunk 3]
```diff
@@ -29,23 +29,23 @@
...
     "eslint-plugin-node": "^11.1.0",
     "eslint-plugin-prettier": "^3.3.1",
-    "jest": "^27.5.1",
+    "jest": "^28.1.3",
     "jest-it-up": "^2.0.2",
     "prettier": "^2.2.1",
     "prettier-plugin-packagejson": "^2.2.17",
     "rimraf": "^3.0.2",
-    "ts-jest": "^27.1.4",
+    "ts-jest": "^28.0.7",
```
Comment (Gudahtt) [inline package.json:48]: This version of `ts-jest` requires a minimum TypeScript version of v4.3, but we're still on v4.2

File: package.json (hunk 2 above)
Comment (Gudahtt) [reply package.json:40]: This is done in #114

File: package.json (hunk 3 above)
Comment (Gudahtt) [reply package.json:48]: This was updated in #116

**Review by mcmire — APPROVED** [review]
LGTM!

**Review by Gudahtt — APPROVED** [review]
LGTM!

**Pull request merged**